Uniswap/interface
# fix: reset tokens when defaultTokens is updated

This is one potential solution to the bug.

Right now, when the `defaultTokens` is updated without using one of the Widget's callback functions, this will not update the Widget's state. Instead, this fix will allow the state to reset if the defaultTokens ever changes while the Widget is being shown.

The downside is this can lead to resetting the state accidentally. For example, if the user had changed the outputToken and the defaultTokens also changed. The only case I can think of right now is the URL changing on token details, which seems like a fair case to reset on.

https://uniswaplabs.atlassian.net/browse/WEB-2921

Test:
- Go to app.uniswap.com.
- Go to the UNI token page via search.
- Go to USDC token page via search.
- Hit back button. UNI token should be shown in widget, not USDC.
